tests/e2e/pubsub/pubsub_test.go

```diff
@@ -203,10 +228,59 @@ func TestPubSub(t *testing.T) {
 	_, err = utils.HTTPGetNTimes(subscriberExternalURL, numHealthChecks)
 	require.NoError(t, err)
 
-	t.Run("pubsubtest1", func(t *testing.T) {
-		sentMessages := sendToPublishApp(t, publisherExternalURL)
+	t.Run("pubsub happy path", func(t *testing.T) {
+		sentMessages := testPublish(t, publisherExternalURL)
 
 		time.Sleep(5 * time.Second)
 		validateMessagesReceivedBySubscriber(t, subscriberExternalURL, sentMessages)
 	})
+
+	t.Run("pubsub no topic ", func(t *testing.T) {
+		testPublishWithoutTopic(t, publisherExternalURL)
+	})
+
+	t.Run("pubsub subscriber error", func(t *testing.T) {
```

Comment:
this is duplicated code. can we use table based test pattern ? I would create new test method for redelivery test and put these sub-tests to new method.